lib/practice_exercises.rb

```diff
@@ -1,13 +1,29 @@
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 
-# Time Complexity: ?
-# Space Complexity: ?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string.downcase!
+  string.gsub!(/[^a-z0-9\s]/i, '')    #removes all non-alphanumeric chars
+  string.gsub!(/\s+/, '')    #removes spaces
+
+  (string.size/2).times.all? do |index|
+    string[index] == string[-index-1]  #returns true or false
+  end
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n)
+# Space Complexity: O(n)
+
 def longest_prefix(strings)
```

Comment:
👍  Very clever

Because this loops through the array of strings and the number of characters in the smallest string, this has O(n + m) time complexity, or O(n) if the strings are all limited in size.